esp/services/ws_workunits/ws_wudetails.hpp

```diff
@@ -0,0 +1,126 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2017 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef _ESPWIZ_ws_wudetails_HPP__
+#define _ESPWIZ_ws_wudetails_HPP__
+
+#include "jlib.hpp"
+
+class CWUDetails
+{
+    Owned<IConstWorkUnit> workunit;
+    StringBuffer wuid;
+    IConstWUScopeFilter & scopeFilter;
+    IConstWUAttributeToReturn & attribsToReturn;
+    IConstWUScopeOptions & scopeOptions;
+    IConstWUAttributeOptions & attribOptions;
+    IConstWUNestedFilter & nfilter;
+
+    class CAttributeFilter: public CInterface, implements IInterface
+    {
+        StatisticKind statisticKind;
+        unsigned __int64 exactValue;
+        unsigned __int64 minValue;
+        unsigned __int64 maxValue;
+        bool bExactValue;
+        bool bMinValue;
+        bool bMaxValue;
+    public:
+        IMPLEMENT_IINTERFACE;
+
+        CAttributeFilter(){};
```

Comment:
Doesn't initialise any of the members - all new code should have default initialisers on any class members.